include/ignition/gazebo/gui/GuiEvents.hh

```diff
@@ -235,6 +238,28 @@ namespace events
     /// \brief Private data pointer
     IGN_UTILS_IMPL_PTR(dataPtr)
   };
+
+  /// \brief Event that notifies a visual plugin is to be loaded
+  class IGNITION_GAZEBO_GUI_VISIBLE VisualPlugins: public QEvent
+  {
+    /// \brief Constructor
+    /// \param[in] _entity Visual entity id
+    /// \param[in] _plugins SDF plugin object
+    public: explicit VisualPlugins(ignition::gazebo::Entity _entity,
+                const sdf::Plugins &_plugins);
+
+    /// \brief Get the entity to load the visual plugin for
+    public: ignition::gazebo::Entity Entity() const;
+
+    /// \brief Get the SDF Plugin of the visual plugin
+    public: const sdf::Plugins &Plugins() const;
+
+    static const QEvent::Type kType = QEvent::Type(QEvent::User + 8);
```

Comment:
The type number needs to be unique across the event types

```suggestion
    static const QEvent::Type kType = QEvent::Type(QEvent::User + 9);
```

Reply:
Done in [afd30fb](https://github.com/gazebosim/gz-sim/pull/1352/commits/afd30fbb79c7c81cf9d7bb056179fa6cd92a0d94)